test/Confluent.Kafka.IntegrationTests/Tests/AdminClient_NullReferenceChecks.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016-2019 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+#pragma warning disable xUnit1026
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Confluent.Kafka.Admin;
+using Xunit;
+
+
+namespace Confluent.Kafka.IntegrationTests
+{
+    public partial class Tests
+    {
+        /// <summary>
+        ///     Tests the validation of input parameter to guard against uncaught segfaults.
+        /// </summary>
+        [Theory, MemberData(nameof(KafkaParameters))]
+        public void AdminClient_NullReferenceChecks(string bootstrapServers)
+        {
+            LogToFile("start AdminClient_NullReferenceChecks");
+            var topicName1 = Guid.NewGuid().ToString();
+            string nullTopic = null;
+
+            // test creating a null topic throws a related exception
+            using (var producer = new ProducerBuilder<Null, Null>(new ProducerConfig { BootstrapServers = bootstrapServers }).Build())
+            using (var adminClient = new DependentAdminClientBuilder(producer.Handle).Build())
+            {
+                try
+                {
+                    adminClient.CreateTopicsAsync(new TopicSpecification[] { new TopicSpecification { Name = nullTopic, NumPartitions = 1, ReplicationFactor = 1 } }).Wait();
+                    Assert.True(false, "Expected exception.");
+                }
+                catch (KafkaException ex)
```

Comment:
This is now inconsistent with `CreatePartitionsAsync`... I think we should change `CreateTopicsAsync` to explicitly check for null topics and throw `ArgumentException` not `KafkaException`. This is a breaking change, but minor enough that I think we should do it.

Feel free to audit the rest of the api for `KafkaException` where `ArgumentException` or derivative more appropriate. I think this approach to exceptions is best - not using `KafkaException` for argument related exceptions, or perhaps other very standard scenarios of which I can't think of any at the moment. But use it for everything else. Welcome feedback on that.

Reply:
Probably common knowledge, but I was curious, so I tracked down the reason for the different exception types. The root cause is that the CreatePartitions with null topic causes a segfault (so I had wrapped it in an exception.  CreateTopic on the other hand just returns a null pointer if the topic is null, so we were already wrapping it in a Kafka exception.

Reply:
Implemented as ArgumentException for both, with a test case to help keep the exceptions consistent between those two 'Create...' methods.